src/main/php/xp/scriptlet/Develop.class.php

```diff
@@ -32,23 +35,49 @@ public function toString() {
   /**
    * Serve requests
    *
-   * @param  xp.scriptlet.WebLayout $layout
+   * @param  string $source
    * @param  string $profile
-   * @param  string $webroot
-   * @param  string $docroot
+   * @param  io.Path $webroot
+   * @param  io.Path $docroot
+   * @param  string[] $config
    */
-  public function serve(WebLayout $layout, $profile, $webroot, $docroot) {
+  public function serve($source, $profile, $webroot, $docroot, $config) {
     $runtime= Runtime::getInstance();
     $startup= $runtime->startupOptions();
     $backing= typeof($startup)->getField('backing')->setAccessible(true)->get($startup);
 
+    // PHP doesn't start with a nonexistant document root
+    if (!$docroot->exists()) {
+      $docroot= getcwd();
+    }
+
     // Start `php -S`, the development webserver
     $arguments= ['-S', $this->host.':'.$this->port, '-t', $docroot];
     $options= newinstance(RuntimeOptions::class, [$backing], [
       'asArguments' => function() use($arguments) {
         return array_merge($arguments, parent::asArguments());
       }
     ]);
+    $options->withSetting('user_dir', $source.PATH_SEPARATOR.implode(PATH_SEPARATOR, $config));
+
+    // Pass classpath
+    $cp= [];
+    foreach (ClassLoader::getLoaders() as $delegate) {
+      if ($delegate instanceof FileSystemClassLoader || $delegate instanceof ArchiveClassLoader) {
+        $cp[]= $delegate->path;
+      }
+    }
+    $options->withClassPath($cp);
+
```

Comment:
This is the essence of this commit - really, `lang.RuntimeOptions` should be handling this correctly!